app/src/main/java/cm/aptoide/pt/wallet/WalletInstallManager.kt

```diff
@@ -14,4 +33,70 @@ class WalletInstallManager(val configuration: WalletInstallConfiguration,
     }.onErrorReturn { null }
 
   }
+
+  fun shouldShowRootInstallWarningPopup(): Boolean {
+    return installManager.showWarning()
+  }
+
+  fun allowRootInstall(answer: Boolean) {
+    installManager.rootInstallAllowed(answer)
+  }
+
+  fun downloadApp(walletApp: WalletApp): Completable {
+    return Observable.just(
+        downloadFactory.create(
+            downloadStateParser.parseDownloadAction(DownloadModel.Action.INSTALL),
+            walletApp.appName,
+            walletApp.packageName,
+            walletApp.md5sum, walletApp.icon, walletApp.versionName, walletApp.versionCode,
+            walletApp.path, walletApp.pathAlt, walletApp.obb,
+            false, walletApp.size))
+        .flatMapSingle { download ->
+          moPubAdsManager.adsVisibilityStatus.doOnSuccess { responseStatus ->
```

Comment:
Forgot to flag this in the review. I think we should change this to getAdsVisibilityStatus(), since it's not a class field of MoPubAdsManager. The IDE will complain a bit, but I think it's better.